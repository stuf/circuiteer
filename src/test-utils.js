import { render as renderRtl } from '@testing-library/react';
import { format } from 'util';
import { Provider } from 'react-redux';
import { preloadedState } from './core/preloaded';
import * as reducers from './state';
import { configureStore } from '@reduxjs/toolkit';

import './test/index';

//

export function render(vdom, { initialState, store, ...renderOptions } = {}) {
  const rootStore =
    store ||
    configureStore({
      preloadedState: initialState || preloadedState,
      reducer: reducers,
    });

  function Wrapper({ children }) {
    return <Provider store={rootStore}>{children}</Provider>;
  }

  return renderRtl(vdom, { wrapper: Wrapper, ...renderOptions });
}

//

export function suppressPropTypeWarnings() {
  const err = console.error.bind(console.error);

  beforeAll(() => {
    console.error = (msg, ...rest) => {
      const errorMsg = format(msg, ...rest);
      if (!errorMsg.toString().includes('Warning: Failed prop type')) {
        err(errorMsg);
      }
    };
  });

  afterAll(() => {
    console.error = err;
  });
}
