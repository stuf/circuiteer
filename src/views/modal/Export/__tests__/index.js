import { render } from 'test-utils';
import { ExportModal } from '../index';

test('ExportModal', () => {
  const { container } = render(<ExportModal />);

  expect(container).toMatchSnapshot();
});
