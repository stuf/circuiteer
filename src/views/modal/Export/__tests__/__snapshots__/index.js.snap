// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExportModal 1`] = `<div />`;
